Source/WebCore/editing/FrameSelection.cpp

```diff
@@ -1893,7 +1893,9 @@ Color CaretBase::computeCaretColor(const RenderStyle& elementStyle, const Node*
         return { };
     return elementStyle.colorResolvingCurrentColor(elementStyle.caretColor());
 #elif HAVE(REDESIGNED_TEXT_CURSOR)
-    if (elementStyle.hasAutoCaretColor() && !elementStyle.hasExplicitlySetColor()) {
+    auto appUsesCustomAccentColor = node && node->document().page() && node->document().page()->appUsesCustomAccentColor();
```

Comment:
Doesn't this need to be behind `HAVE(APP_ACCENT_COLORS)`, so the the Catalyst build doesn't break?